src/modules/flight_mode_manager/tasks/AutoMapper/FlightTaskAutoMapper.cpp

```diff
@@ -77,6 +77,16 @@ bool FlightTaskAutoMapper::update()
 		break;
 
 	case WaypointType::land:
+	        if (_type_previous != _type) {
+			// Reset the trajectory current position when initializing landing.
+			// This is important when time stretching is disabled for landing trajectory,
+			// when the drone is blown away from its current setpoint by strong winds.
+			// This reduces that chance that when the drone descends to lower altitudes
+			// with potentially less wind, and moves towards its previous trajectory setpoint.
+			// TODO Should be set by a common parameter as disable time stretch for landing.
+			// TODO evaluate if needed to reset position again during landing if drift is too high
+			_ekfResetHandlerPositionXY();
```

Comment:
After thinking more about this, I'm not sure if this reset is actually needed since `Land::on_active()` should update the `position_setpoint_triplet` when entering land mode. 
Did you see any significantly different behavior in the simulator with or without this reset?

For context, see `Land::on_active()` in https://github.com/aviant-tech/PX4-Autopilot/blob/957ce0d2f77a9533a11bf5a2997c2525c2bac2db/src/modules/navigator/land.cpp#L50
which calls `set_land_item` https://github.com/aviant-tech/PX4-Autopilot/blob/957ce0d2f77a9533a11bf5a2997c2525c2bac2db/src/modules/navigator/mission_block.cpp#L695